dipy/denoise/localpca.py

```diff
@@ -1,29 +1,34 @@
 import numpy as np
-import scipy as sp
-import scipy.linalg as sla
-from warnings import warn
-
-# Try to get the SVD through direct API to lapack:
 try:
-    from scipy.linalg.lapack import sgesvd as svd
+    from scipy.linalg.lapack import dgesvd as svd
     svd_args = [1, 0]
-# If you have an older version of scipy, we fall back
-# on the standard scipy SVD API:
+    # If you have an older version of scipy, we fall back
+    # on the standard scipy SVD API:
 except ImportError:
     from scipy.linalg import svd
     svd_args = [False]
+from scipy.linalg import eigh
 
 
-def localpca(arr, sigma, patch_radius=2, tau_factor=2.3, out_dtype=None):
+def localpca(arr, sigma, mask=None, pca_method='eig', patch_radius=2,
+             tau_factor=2.3, out_dtype=None):
     r"""Local PCA-based denoising of diffusion datasets.
 
     Parameters
     ----------
     arr : 4D array
         Array of data to be denoised. The dimensions are (X, Y, Z, N), where N
         are the diffusion gradient directions.
+    mask : 3D boolean array
+        A mask with voxels that are true inside the brain and false outside of
+        it.
```

Comment:
I'm still missing an explanation of what the mask does here: it denoises within the True part and returns *zeros* outside of those voxels, right?